File: packages/device_info/lib/src/flutter_config/src/models/src/general_info/general_info.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'general_info.freezed.dart';
part 'general_info.g.dart';

@freezed
class GeneralInfo with _$GeneralInfo {
  const GeneralInfo._();

  const factory GeneralInfo({
    required int usedSpace,
    required int totalSpace,
    required int usedMemory,
    required int totalMemory,
    required int freeMemory,
    required int battery,
  }) = _GeneralInfo;

  factory GeneralInfo.fromJson(Map<String, dynamic> json) =>
      _$GeneralInfoFromJson(json);
}

//flutter pub run build_runner watch --delete-conflicting-outputs

File: packages/device_info/android/src/main/kotlin/com/rocket/device/info/models/GeneralInfo.kt
package com.rocket.device.info.models

import org.json.JSONObject

class GeneralInfo(val generalInfo: Map<String, Any>) {
    val usedSpace:Int by generalInfo
    val totalSpace:Int by generalInfo
    val usedMemory:Int by generalInfo
    val totalMemory:Int by generalInfo
    val freeMemory:Int by generalInfo
    val battery:Int by generalInfo

    override fun toString(): String {
        return "${JSONObject(generalInfo)}"
    }
}

